app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -191,25 +183,19 @@ class TransactionDetailsViewModel
                 )
             }.onSuccess {
                 txDetails = it
+                val newExecutionInfo = txDetails?.detailedExecutionInfo as? DetailedExecutionInfo.MultisigExecutionDetails
+                    ?: throw MissingCorrectExecutionDetailsException
```

Comment:
So in case you get `ExecutionDetails` that aren't from the type you are expecting, it throws? Couldn't we leverage types better here? Or, if we are in branch of code in which we know for sure the type, just simply do a hard cast?